pkg/client/client.go

```diff
@@ -1244,3 +1322,80 @@ func check(dep pkg.Dependency, newDepPath string) bool {
 
 	return dep.Sum == sum
 }
+
+func FindSource[K comparable, T any](g graph.Graph[K, T]) ([]K, error) {
```

Comment:
Move graph related functions to graph.go or dep_graph.go

Reply:
Done !